### ISSUE TITLE
allow zero amount payment request

### ISSUE THREAD

**Issue by mr-highball** [btcontract/lnwallet#68]
First, you have an excellent wallet that just "works".

Issue
--
One issue I'm seeing is the requirement to specify an amount in the payment request. According to bolt 11, an amount should be optional, and is useful for things such as donations
[rfc entry](https://github.com/lightningnetwork/lightning-rfc/blob/4aabe6052395269f82e20b06f1e6b717c343f7d9/11-payment-encoding.md#requirements-1)

Steps To Reproduce
--

1. tap action button
1. tap "Receive funds"
1. tap "Lightning payment"
1. leave amount empty or type 0
1. attempt to generate invoice

**Comment by mr-highball** [comment]
possibly related to #58

**Comment by btcontract** [comment]
I've decided to explicitly disallow a "donation case" because it has some non-obvious but serious issues:

1. LN payment request is disposable: payer's wallet will reject a second outgoing payment for the same request and payee's wallet will fail a second incoming payment for the same request. Payment request reusing is actively discouraged because a preimage gets revealed after the very first payment which would let routing nodes to "fulfill" the subsequent transfers without routing them to payee.

2. When fulfilling such a payment request a payer may provide an amount which is higher than what payee can currently receive which would result in a failure so an exact sum better always be communicated beforehand.

In conclusion, receiving payments is really tough in LN right now. One has to be online and better provide an exact amount and do it once per payment. This does not bode well with donations since people expect a donation income to be passive and "always there", not as something one has to do each time exactly when someone wants to donate.

**Comment by mr-highball** [comment]
Thank you for elaborating on this, all makes sense 👍
